app/controllers/my-projects/assignment/recommendations/add.js

```diff
@@ -263,6 +269,16 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
     this.set('modalOpen', false);
   }
 
+  // adds the uploaded file to each disposition queue
+  @action
+  async addFileToDispositionQueues(file) {
+    this.dispositions.map(async (disposition) => {
+      // use this private/undocument .blob property to duplicate the file
+      const newFile = await File.fromBlob(file.blob);
+      this.queuesByDisposition[disposition.id].push(newFile);
+    });
```

Comment:
In this case, `map` is for iterating over an array and _returning a new array_ based on the return value. I think it's unclear here when you use map because you're not using the return value, and your callback function implicitly returns a promise (though it doesn't explicitly return something inside).

Reply:
Hmm yeah you're right. I'm not sure what I would return from the action though, since nothing would ever use the action's return value. Maybe I'll at least make the map's return of a Promise more explicit, though.

Reply:
@allthesignals updated, take a look if you get a free moment!